src/net_processing.cpp

```diff
@@ -4424,10 +4424,10 @@ void PeerManagerImpl::MaybeSendPing(CNode& node_to, Peer& peer, std::chrono::mic
     }
 
     if (pingSend) {
-        uint64_t nonce = 0;
-        while (nonce == 0) {
-            GetRandBytes({(unsigned char*)&nonce, sizeof(nonce)});
-        }
+        uint64_t nonce;
+        do {
+            nonce = GetRand<uint64_t>();
```

Comment:
I wonder if it makes sense to use a single FastRandomContext member for all entropy drawn inside `PeerManagerImpl`, instead of creating a new context for each call.